caas/kubernetes/provider/resources/pod.go

```diff
@@ -112,12 +112,7 @@ func (p *Pod) Delete(ctx context.Context) error {
 	err := p.client.Delete(ctx, p.Name, metav1.DeleteOptions{
 		PropagationPolicy: k8sconstants.DefaultPropagationPolicy(),
 	})
-	if k8serrors.IsNotFound(err) {
```

Comment:
I see that we're now handling the k8s not found case in the applier process(). Given that's the only current call site for Delete(), it works. We'll need to be careful thought that other future work which might call Delete() doesn't forgot to do the same.

We also have the Get() method mapping k8s not found to juju errors NotFound. It seems suboptimal to not adopt a uniform approach to error mapping in the apis on the same object. We should consider doing the same mapping here too.